src/SFML/System/MemoryInputStream.cpp

```diff
@@ -36,19 +36,19 @@ namespace sf
 void MemoryInputStream::open(const void* data, std::size_t sizeInBytes)
 {
     m_data   = static_cast<const std::byte*>(data);
-    m_size   = static_cast<std::int64_t>(sizeInBytes);
+    m_size   = sizeInBytes;
     m_offset = 0;
 }
 
 
 ////////////////////////////////////////////////////////////
-std::int64_t MemoryInputStream::read(void* data, std::int64_t size)
+std::optional<std::size_t> MemoryInputStream::read(void* data, std::size_t size)
 {
     if (!m_data)
-        return -1;
+        return std::nullopt;
 
-    const std::int64_t endPosition = m_offset + size;
-    const std::int64_t count       = endPosition <= m_size ? size : m_size - m_offset;
+    const std::size_t endPosition = m_offset + size;
+    const std::size_t count       = endPosition <= m_size ? size : m_size - m_offset;
```

Comment:
Changing this from signed to unsigned arithmetic gives me pause. I want to be sure I didn't accidentally change how this expression evaluates.

Reply:
It looks ok, but IMO it would be more readable with `std::min(m_size, m_offset + size)` to get the end position, then computing count as `end - m_offset`.